cpp/src/traversal/sssp_impl.cuh

```diff
@@ -69,23 +69,113 @@ struct pred_op_t {
     auto new_distance = src_val + w;
     auto threshold    = cutoff;
     if (vertex_partition.in_local_vertex_partition_range_nocheck(dst)) {
-      auto local_vertex_offset =
-        vertex_partition.local_vertex_partition_offset_from_vertex_nocheck(dst);
-      auto old_distance = *(distances + local_vertex_offset);
+      auto v_offset     = vertex_partition.local_vertex_partition_offset_from_vertex_nocheck(dst);
+      auto old_distance = distances[v_offset];
       threshold         = old_distance < threshold ? old_distance : threshold;
     }
     if (new_distance >= threshold) { push = false; }
     return push;
   }
 };
 
+template <typename weight_t>
+weight_t compute_subpartition_start(weight_t old_near_far_threshold,
+                                    size_t subpartition_idx,
+                                    size_t num_subpartitions,
+                                    weight_t delta)
+{
+  return old_near_far_threshold + delta * (static_cast<weight_t>(subpartition_idx) /
+                                           static_cast<weight_t>(num_subpartitions));
+}
+
+template <typename vertex_t, typename weight_t, bool multi_gpu>
+std::tuple<size_t, size_t> compute_new_near_nar_partition_range(
```

Comment:
Should this be `compute_new_near_far_partition_range`?

Reply:
Oh, this actually should be near_near... (e) was missing here :-(